integration_tests/gettxs_test.go

```diff
@@ -0,0 +1,165 @@
+package integration_tests
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/getAlby/lndhub.go/controllers"
+	"github.com/getAlby/lndhub.go/lib"
+	"github.com/getAlby/lndhub.go/lib/responses"
+	"github.com/getAlby/lndhub.go/lib/service"
+	"github.com/getAlby/lndhub.go/lib/tokens"
+	"github.com/getAlby/lndhub.go/lnd"
+	"github.com/go-playground/validator/v10"
+	"github.com/labstack/echo/v4"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+type GetTxTestSuite struct {
+	TestSuite
+	Service                  *service.LndhubService
+	fundingClient            *lnd.LNDWrapper
+	userLogin                controllers.CreateUserResponseBody
+	userToken                string
+	invoiceUpdateSubCancelFn context.CancelFunc
+}
+
+type GetOutgoingInvoiceResponseTest struct {
+	RHash           interface{} `json:"r_hash"`
+	PaymentHash     interface{} `json:"payment_hash"`
+	PaymentPreimage string      `json:"payment_preimage"`
+	Value           int64       `json:"value"`
+	Fee             int64       `json:"fee"`
+	Timestamp       int64       `json:"timestamp"`
+	Memo            string      `json:"memo"`
+}
+
+type GetIncomingInvoiceResponseTest struct {
+	RHash          interface{} `json:"r_hash"`
+	PaymentHash    interface{} `json:"payment_hash"`
+	PaymentRequest string      `json:"payment_request"`
+	Description    string      `json:"description"`
+	PayReq         string      `json:"pay_req"`
+	Timestamp      int64       `json:"timestamp"`
+	Type           string      `json:"type"`
+	ExpireTime     int64       `json:"expire_time"`
+	Amount         int64       `json:"amt"`
+	IsPaid         bool        `json:"ispaid"`
+}
+
+func (suite *GetTxTestSuite) SetupSuite() {
+	lndClient, err := lnd.NewLNDclient(lnd.LNDoptions{
+		Address:     lnd2RegtestAddress,
+		MacaroonHex: lnd2RegtestMacaroonHex,
+	})
+	if err != nil {
+		log.Fatalf("Error setting up funding client: %v", err)
+	}
+	suite.fundingClient = lndClient
+
+	svc, err := LndHubTestServiceInit()
+	if err != nil {
+		log.Fatalf("Error initializing test service: %v", err)
+	}
+	users, userTokens, err := createUsers(svc, 1)
+	if err != nil {
+		log.Fatalf("Error creating test users %v", err)
+	}
+	// Subscribe to LND invoice updates in the background
+	// store cancel func to be called in tear down suite
+	ctx, cancel := context.WithCancel(context.Background())
+	suite.invoiceUpdateSubCancelFn = cancel
```

Comment:
is there a better way to close this subscription? this test suite and following both subscribe, i was getting weird issues until i noticed we need to close this in tear down suite
if this is ok way, i would do the same for other tests